config/routes.rb

```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  get '/posts/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/posts/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  get '/users', to: 'users#index', as: 'users'
+  get '/users/:id', to: 'users#show', as: 'user'  
```

Comment:
You did well implementing these milestone requirements correctly. 

- [ ] [OPTIONAL] I see that you coded the routes explicitly which is okay only that this is prone to error especially as your app begins to grow large. This method of declaring routes becomes very difficult to maintain or even follow. Rails have an inbuilt method `resource` that you can use easily to avoid possible errors. 

```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
``` 

Resource routing allows you to quickly declare all of the common routes for a given resourceful controller. A single call to resources can declare all of the necessary routes for your index , show , new , edit , create , update , and destroy actions.

Learn about this method [here](https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default) and [here](https://launchschool.com/books/demystifying_rails/read/routes_and_resources). I recommend you begin with the second link.